src/MOI_wrapper.jl

```diff
@@ -940,25 +1218,83 @@ function MOI.jacobian_structure(model::Optimizer)
     return J
 end
 
+function _eval_constraint_jacobian(
+    values::AbstractVector,
+    offset::Int,
+    x::AbstractVector,
+    f::MOI.VectorOfVariables,
+    s::VectorNonlinearOracle,
+)
+    for i in 1:s.input_dimension
+        s.x[i] = x[f.variables[i].value]
+    end
+    nnz = length(s.jacobian_structure)
+    s.eval_jacobian(view(values, offset .+ (1:nnz)), s.x)
+    return offset + nnz
+end
+
 function MOI.eval_constraint_jacobian(model::Optimizer, values, x)
     offset = MOI.eval_constraint_jacobian(model.qp_data, values, x)
-    nlp_values = view(values, offset:length(values))
+    offset -= 1  # .qp_data returns one-indexed offset
+    for (f, s) in model.vector_nonlinear_oracle_constraints
+        offset = _eval_constraint_jacobian(values, offset, x, f, s)
+    end
+    nlp_values = view(values, (offset+1):length(values))
     MOI.eval_constraint_jacobian(model.nlp_data.evaluator, nlp_values, x)
     return
 end
 
 ### Eval_H_CB
 
+function _hessian_lagrangian_structure(
+    ret::AbstractVector,
+    f::MOI.VectorOfVariables,
+    s::VectorNonlinearOracle,
+)
+    for (i, j) in s.hessian_lagrangian_structure
+        push!(ret, (f.variables[i].value, f.variables[j].value))
+    end
+    return
+end
+
 function MOI.hessian_lagrangian_structure(model::Optimizer)
     H = MOI.hessian_lagrangian_structure(model.qp_data)
+    for (f, s) in model.vector_nonlinear_oracle_constraints
+        _hessian_lagrangian_structure(H, f, s)
+    end
     append!(H, MOI.hessian_lagrangian_structure(model.nlp_data.evaluator))
     return H
 end
 
+function _eval_hessian_lagrangian(
+    H::AbstractVector,
+    H_offset::Int,
+    x::AbstractVector,
+    μ::AbstractVector,
+    μ_offset::Int,
+    f::MOI.VectorOfVariables,
+    s::VectorNonlinearOracle,
+)
+    for i in 1:s.input_dimension
+        s.x[i] = x[f.variables[i].value]
+    end
+    H_nnz, μ_nnz = length(s.hessian_lagrangian_structure), s.output_dimension
+    H_view = view(H, H_offset .+ (1:H_nnz))
+    μ_view = view(μ, μ_offset .+ (1:μ_nnz))
```

Comment:
Nitpick: `μ_nnz` is a little confusing here.
```suggestion
    H_nnz = length(s.hessian_lagrangian_structure)
    H_view = view(H, H_offset .+ (1:H_nnz))
    μ_view = view(μ, μ_offset .+ (1:s.output_dimension))
```